sway/commands/input/xkb_switch_layout.c

```diff
@@ -13,6 +13,26 @@ static void switch_layout(struct wlr_keyboard *kbd, xkb_layout_index_t idx) {
 		kbd->modifiers.latched, kbd->modifiers.locked, idx);
 }
 
+static xkb_layout_index_t get_current_layout_index(struct wlr_keyboard *kbd) {
+	xkb_layout_index_t num_layouts = xkb_keymap_num_layouts(kbd->keymap);
+	xkb_layout_index_t layout_idx;
+	for (layout_idx = 0; layout_idx < num_layouts; layout_idx++) {
+		if (xkb_state_layout_index_is_active(kbd->xkb_state,
+				layout_idx, XKB_STATE_LAYOUT_EFFECTIVE)) {
+			break;
+		}
+	}
+	return layout_idx;
+}
+
+static void switch_layout_relative(struct wlr_keyboard *kbd, int dir) {
+	xkb_layout_index_t num_layouts = xkb_keymap_num_layouts(kbd->keymap);
+	xkb_layout_index_t idx = get_current_layout_index(kbd);
+	wlr_keyboard_notify_modifiers(kbd, kbd->modifiers.depressed,
+		kbd->modifiers.latched, kbd->modifiers.locked,
+		((idx + num_layouts) + dir) % num_layouts);
```

Comment:
Style: unnecessary parentheses here

Reply:
This was to prevent potential overflow with `idx=0`, `dir=-1` but it probably doesn't really matter.